### PR TITLE
[MM-20620] Handle trailing slash when building marketplace api path

### PR DESCRIPTION
**Pull request by fm2munsh**
#### Summary
- Update Client#buildURL to normalize both the incoming path and the existing marketplace url by removing leading & trailing slashes and then adding a single slash in between the two strings
- Also add some unit tests for `buildURL`

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/13271
JIRA: https://mattermost.atlassian.net/browse/MM-20620

### CODE REVIEW

File: services/marketplace/client_test.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+		base     string
+		path     string
+		expected string
+	}{
+		"Base url with trailing slash and path with leading slash": {
+			base:     "https://api.integrations.mattermost.com/",
+			path:     "/api/v1/plugins",
+			expected: "https://api.integrations.mattermost.com/api/v1/plugins",
+		},
+		"Base url with without trailing slash and path with leading slash": {
```
Comment (marianunez) [inline services/marketplace/client_test.go]:
```suggestion
		"Base url without trailing slash and path with leading slash": {
```

File: services/marketplace/client_test.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+			base:     "https://api.integrations.mattermost.com",
+			path:     "/api/v1/plugins",
+			expected: "https://api.integrations.mattermost.com/api/v1/plugins",
+		},
+		"Base url with without trailing slash and path without leading slash": {
```
Comment (marianunez) [inline services/marketplace/client_test.go]:
```suggestion
		"Base url without trailing slash and path without leading slash": {
```

**Comment by hanzei** [comment]
Thanks for fixing this @fm2munsh :+1:

**Pull request merged**